cmd/kubeadm/app/cmd/reset.go

```diff
@@ -131,16 +138,16 @@ func NewReset(in io.Reader, ignorePreflightErrors sets.String, forceReset bool,
 }
 
 // Run reverts any changes made to this host by "kubeadm init" or "kubeadm join".
-func (r *Reset) Run(out io.Writer, client clientset.Interface) error {
+func (r *Reset) Run(out io.Writer, client clientset.Interface, cfg *kubeadmapi.InitConfiguration) error {
 	var dirsToClean []string
 	// Only clear etcd data when using local etcd.
 	etcdManifestPath := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ManifestsSubDirName, "etcd.yaml")
 
 	klog.V(1).Infof("[reset] checking for etcd config")
-	etcdDataDir, err := getEtcdDataDir(etcdManifestPath, client)
+	etcdDataDir, err := getEtcdDataDir(etcdManifestPath, cfg)
 	if err == nil {
 		dirsToClean = append(dirsToClean, etcdDataDir)
-		if err := removeEtcdMember(client); err != nil {
+		if err := etcdphase.RemoveStackedEtcdMemberFromCluster(client, cfg); err != nil {
```

Comment:
I don't know how to deal with this code, because `emoveStackedEtcdMemberFromCluster` should use `client` to  connect cluster and get etcd client. So i add new parameters that name `cfg` in `func (r *Reset) Run()`.

Reply:
this seems good to me on a quick look. let's see what the others think.

Reply:
Thanks for your review.